py/desisim/simexp.py

```diff
@@ -480,9 +485,11 @@ def simulate_spectra(wave, flux, fibermap=None, obsconditions=None,
         lrgs=(source_types=="lrg")
         bgss=(source_types=="bgs")
 
-        if np.sum(lrgs)>0 or np.sum(elgs)>0 or np.sum(bgss)>0 :
-            log.warning("the half light radii are fixed here (and not magnitude or redshift dependent)")
-
+        if np.sum(lrgs)>0 or np.sum(elgs)>0:
+            log.warning("the half light radii are fixed here for LRGs and ELGs (and not magnitude or redshift dependent)")
+        if np.sum(bgss)>0 and source_redshift is None:
+            log.warning("the half light radii are fixed here for BGS (as redshifts weren't supplied)")
+            
         # BGS parameters based on SDSS main sample, in g-band
         # see analysis from J. Moustakas in
         # https://github.com/desihub/desitarget/blob/bgs-properties/doc/nb/bgs-morphology-properties.ipynb
```

Comment:
Can you update this notebook link to https://github.com/desihub/desitarget/blob/master/doc/nb/bgs-morphology-properties.ipynb please?